### DISCUSSION TITLE
TOTP setup failed with: The user session elevation did not have a matching IP

### DISCUSSION THREAD

**Discussion by justason** [authelia/authelia#8221] [Q&A] [answered]
Using Authelia + traefik docker container setup, pretty much basic. Trying to secure access to whoami container using two-factor authentication.
Getting TOTP failed with an error:
time="2024-11-10T20:22:06+02:00" level=warning msg="The user session elevation did not have a matching IP. 
It will be destroyed and the users access will be forbidden." 
expected_ip=172.71.103.178 method=POST path=/api/secondfactor/totp/register remote_ip=162.***.***.***user=*****
And TOTP setup is failed.

Any suggestions are welcome.

**Comment by james-d-elliott** [comment]
With trace logs on do the IP's of the clients match between the start of the elevation request and after?

**Comment by franco820711** [comment, marked as answer]
I have the same issue, I think it's because my Cloudflare proxy passes different IP addresses between the session and TOTP's request IP. I succeeded once, but the second time it didn't work no matter how I tried. I'm still investigating the cause.

**Reply by james-d-elliott** [reply]
Nice that's probably exactly it. You will need to trust the cloudflare IP's as proxy sources. Also keep in mind [this](https://www.authelia.com/integration/proxies/forwarded-headers/#cloudflare).
